pytorch_lightning/callbacks/gpu_stats_monitor.py

```diff
@@ -111,7 +111,14 @@ def on_train_start(self, trainer, pl_module) -> None:
                 f' since gpus attribute in Trainer is set to {trainer.gpus}.'
             )
 
-        self._gpu_ids = ','.join(map(str, trainer.data_parallel_device_ids))
+        self._device_ids = list(trainer.data_parallel_device_ids)
+
+        cuda_visible_devices = os.getenv('CUDA_VISIBLE_DEVICES', default=None)
+        if cuda_visible_devices is None:
+            self._gpu_ids = ','.join(map(str, self._device_ids))
+        else:
+            cuda_visible_devices = cuda_visible_devices.split(',')
+            self._gpu_ids = ','.join(map(cuda_visible_devices.__getitem__, self._device_ids))
```

Comment:
could we define these in `__init__` and then put a comment for each, so it is clear that device_ids is for the selected ids, and gpu_ids are the unmasked ids.

Reply:
These attributes cannot be assigned in `__init__`, because the `Trainner` object have not created at all. They can be assigned in `callback.setup()`. I have added the comments in `setup()` now.

Reply:
Yes, I mean initialize them as empty list in the init.